src/main/java/gregtech/common/metatileentities/multi/electric/MetaTileEntityAssemblyLine.java

```diff
@@ -46,44 +65,106 @@ public MetaTileEntity createMetaTileEntity(IGregTechTileEntity tileEntity) {
-                .aisle("FIF", "RTR", "SAG", "#Y#")
-                .aisle("FIF", "RTR", "GAG", "#Y#").setRepeatable(3, 15)
-                .aisle("FOF", "RTR", "GAG", "#Y#")
+        FactoryBlockPattern pattern = FactoryBlockPattern.start(FRONT, UP, RIGHT)
+                .aisle("FIF", "RTR", "SAG", " Y ")
+                .aisle("FIF", "RTR", "DAG", " Y ").setRepeatable(3, 15)
+                .aisle("FOF", "RTR", "GAG", " Y ")
                 .where('S', selfPredicate())
                 .where('F', states(getCasingState())
                         .or(autoAbilities(false, true, false, false, false, false, false))
-                        .or(abilities(MultiblockAbility.IMPORT_FLUIDS).setMaxGlobalLimited(4)))
-                .where('O', abilities(MultiblockAbility.EXPORT_ITEMS).addTooltips("gregtech.multiblock.pattern.location_end"))
-                .where('Y', states(getCasingState()).or(abilities(MultiblockAbility.INPUT_ENERGY).setMinGlobalLimited(1).setMaxGlobalLimited(3)))
-                .where('I', metaTileEntities(MetaTileEntities.ITEM_IMPORT_BUS[0]))
-                .where('G', states(MetaBlocks.MULTIBLOCK_CASING.getState(BlockMultiblockCasing.MultiblockCasingType.GRATE_CASING)))
+                        .or(orderedFluidPredicate()))
+                .where('O', abilities(MultiblockAbility.EXPORT_ITEMS)
+                        .addTooltips("gregtech.multiblock.pattern.location_end"))
+                .where('Y', states(getCasingState())
+                        .or(abilities(MultiblockAbility.INPUT_ENERGY)
+                                .setMinGlobalLimited(1)
+                                .setMaxGlobalLimited(3)))
+                .where('I', metaTileEntities(MetaTileEntities.ITEM_IMPORT_BUS[GTValues.ULV]))
+                .where('G', states(getGrateState()))
                 .where('A', states(MetaBlocks.MULTIBLOCK_CASING.getState(BlockMultiblockCasing.MultiblockCasingType.ASSEMBLY_CONTROL)))
                 .where('R', states(MetaBlocks.TRANSPARENT_CASING.getState(BlockGlassCasing.CasingType.LAMINATED_GLASS)))
                 .where('T', states(MetaBlocks.MULTIBLOCK_CASING.getState(BlockMultiblockCasing.MultiblockCasingType.ASSEMBLY_LINE_CASING)))
-                .where('#', any())
-                .build();
+                .where('D', dataHatchPredicate())
+                .where(' ', any());
+        return pattern.build();
+    }
+
+    @Nonnull
+    protected static IBlockState getCasingState() {
+        return MetaBlocks.METAL_CASING.getState(BlockMetalCasing.MetalCasingType.STEEL_SOLID);
+    }
+
+    @Nonnull
+    protected static IBlockState getGrateState() {
+        return MetaBlocks.MULTIBLOCK_CASING.getState(BlockMultiblockCasing.MultiblockCasingType.GRATE_CASING);
+    }
+
+    @Nonnull
+    protected static TraceabilityPredicate orderedFluidPredicate() {
+        // if ordered fluids are enabled, ban multi fluid hatches, otherwise allow all types
```

Comment:
To be consistent, I think we should just (from this point onwards) always ban the multi fluid hatches